### DISCUSSION TITLE
Blocked: NotSameOrigin - While fetching Image of Instagram Post

### DISCUSSION THREAD

**Discussion by harshshah8996** [pgrimaud/instagram-user-feed#204] [Q&A] [answered]
Hello Everyone,

I am using this package to get Instagram Post. I am getting a successful response from API, which has all data like id, caption, thumbnail URL, etc.
But when I tried to use thumbnail or display URL of post's image, I am getting **Blocked: NotSameOrigin** error.

Can you guys help me with how to fix it?

![Screenshot from 2021-04-26 11-40-35](https://user-images.githubusercontent.com/32017961/116036574-87d96f80-a684-11eb-9a06-87ecb71de232.png)

**Comment by pgrimaud** [comment, marked as answer]
Hello,

Weird, it's a CORS error.
Could you send me the link of one picture you tried to fetch.

And headers you set when loading pictures. 

How are you loading pictures? With VueJS I guess, but could you paste the code you wrote?

**Reply by harshshah8996** [reply]
Hello,

URL - https://scontent-lax3-1.cdninstagram.com/v/t51.2885-15/sh0.08/e35/s640x640/177655121_810577946227163_6263465597785604912_n.jpg?tp=1&_nc_ht=scontent-lax3-1.cdninstagram.com&_nc_cat=1&_nc_ohc=dCrz_9uad-AAX8z-Otk&edm=APU89FABAAAA&ccb=7-4&oh=72f147e345d413e00b7b0b2c4f60d4a8&oe=60ACD745&_nc_sid=86f79a

Our use case is just to display posts of users.
So, we get a response by using your package and send a list of URLs of images to the frontend.

The frontend put URL in src of img tag.

If you want to test, you can use username - ndtv

**Reply by pgrimaud** [reply]
Weird, 

When I open your link, I got : 
![image](https://user-images.githubusercontent.com/1866496/116076292-52d51900-a694-11eb-9c96-3b7b147d20ee.png)

`cross-origin-resource-policy: same-origin` (which is the problem)

And when I fetch ndtv account, I got pictures without this cors protection. 

![image](https://user-images.githubusercontent.com/1866496/116076427-7f893080-a694-11eb-9fc9-b6d4e89788c4.png)

How old is the post? Did you store image link into database?

**Reply by harshshah8996** [reply]
Hello,

No, we did not store the image URLs in the database.
Actually, approx 2 weeks before, all image links were working, but now none of them. 

Yes, the issue is because of CORS.
Can you guide me on how to fix this or any other alternative to use image URL?

**Reply by pgrimaud** [reply]
Ok, gotcha. 

The Instagram pictures have parameters which seem to expire after couple of days.

So the best way could be : download this pictures when you fetch feed and posts.

**Reply by harshshah8996** [reply]
Yes, you are right.
But when we tried to get the latest post, we are not able to open the image because of the CORS error.

**Reply by pgrimaud** [reply]
You could download picture with your PHP script, when you fetch medias from user.
